torch/_dynamo/eval_frame.py

```diff
@@ -1201,6 +1202,22 @@ def result_capturing_wrapper(*graph_inputs):
         # optimize_assert since user program may mutate the inputs.
         flat_args, in_spec = pytree.tree_flatten((args, kwargs))
 
+        for arg in flat_args:
+            prim_types = (torch.Tensor, str, int, float, bool)
+            if not isinstance(arg, prim_types):
+                if dataclasses.is_dataclass(arg):
+                    raise RuntimeError(
+                        "Dataclasses are supposed to be pytree nodes to be exportable."
+                        "Please take a look at torch._export.utils.register_dataclass_as_pytree_node "
```

Comment:
depressing that we're suggesting private APIs to people

Reply:
yeah actually we should just publish this API. @tugsbayasgalan

Reply:
It is public [`torch.export.register_dataclass`](https://fburl.com/code/hg2la1lc)